src/map/quest.c

```diff
@@ -321,7 +321,10 @@ static void quest_update_objective(struct map_session_data *sd, const struct mob
 				sd->quest_log[i].count[j] < qi->objectives[j].count &&
 				(qi->objectives[j].level.min == 0 || qi->objectives[j].level.min <= md->level) &&
 				(qi->objectives[j].level.max == 0 || qi->objectives[j].level.max >= md->level) &&
-				(qi->objectives[j].mapid < 0 || qi->objectives[j].mapid == md->bl.m)) {
+				(qi->objectives[j].mapid < 0 || qi->objectives[j].mapid == md->bl.m) &&
+				(qi->objectives[j].mobtype.size == 0 || (qi->objectives[j].mobtype.size & md->status.size) != 0) &&
```

Comment:
Note that `md->status.size` isn't a bitmask! For example, `SZ_SMALL` (or `Size_Small` in scripts/DBs) is defined as 0, making this bitmask comparison always fail for small sized monsters.

If you want this to support bitmasks on the quest db side, then you need to do the comparison for example as `(qi->objectives[j].mobtype.size & (1 << md->status.size)) != 0` (and document to the user how the bits are mapped).

The same goes for elements and races (which are defined in a similar way to the sizes).

Otherwise, if a bit mask isn't necessary, the comparison can be done directly with i.e. `qi->objectives[j].mobtype.size == md->status.size `.

Reply:
I seem to have mixed client and server values for this, for now I removed the bitmask support for these fields.